### PR TITLE
bpo-36583: Do not swallow exceptions in the _ssl module.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue36583

### CODE REVIEW

**Comment by tirkarthi** [comment]
I think the relevant issue is https://bugs.python.org/issue36583 ? 

https://bugs.python.org/issue339827 returns 404 but @bedevere-bot seems to have accepted it. Seems to be known issue : https://github.com/python/bedevere/issues/15

**Comment by serhiy-storchaka** [comment]
Thanks @tirkarthi. I have no idea from where this number came. I just pasted it from the clipboard, but seems something messed.

File: Modules/_ssl.c [hunk 1]
```diff
@@ -3985,22 +3988,44 @@ _ssl__SSLContext_load_verify_locations_impl(PySSLContext *self,
...
         goto error;
     }
 
     /* validata cadata type and load cadata */
     if (cadata) {
-        Py_buffer buf;
-        PyObject *cadata_ascii = NULL;
-
-        if (PyObject_GetBuffer(cadata, &buf, PyBUF_SIMPLE) == 0) {
+        if (PyUnicode_Check(cadata)) {
```
Comment (tiran) [inline Modules/_ssl.c:4007]: Why did you change the order of checks here? What was wrong with the original code?

Comment (serhiy-storchaka) [reply Modules/_ssl.c:4007]: Nothing wrong, but it seems to me that checking for unicode string prior to checking for bytes-like objects is more common. If you prefer, I'll return the former order. This will minimize the diff.
